src/features/archaeologist/discovery.ts

```diff
@@ -38,16 +41,18 @@ export async function initialisePeerDiscovery(browserNode: Libp2p, setArchs: (ar
 
   // Listen for peers connecting
   browserNode.connectionManager.addEventListener('peer:connect', async (evt) => {
-    // TODO: will need to track all connected nodes, and set this value
-    // based on user input
-    // selectedArweaveConn = evt.detail;
-
     const peerId = evt.detail.remotePeer.toString();
+    nodeConnections[peerId] = evt.detail;
     console.log(`Connection established to: ${peerId.slice(peerId.length - idTruncateLimit)}`);
   });
 
   browserNode.pubsub.addEventListener('message', (evt) => {
     const msg = new TextDecoder().decode(evt.detail.data);
+    if (msg.startsWith('signature')) {
+      console.log('signature', msg);
+      return;
+    }
```

Comment:
Once we switch to streaming, this could check the protocol the message is coming in on (maybe in a switch statement) and handle appropriately.